contracts/fee_splitter/src/testing.rs

```diff
@@ -1 +1,144 @@
+    };
+
+    let res = instantiate(deps.as_mut(), env, info, msg).unwrap_err();
+    assert_eq!(res, ContractError::InvalidWeights {});
+}
+
+#[test]
+fn should_send_tokens_in_correct_amount() {
+    let mut deps = mock_coreum_deps(&[]);
+
+    deps.querier.with_balance(&[(
+        &String::from(MOCK_CONTRACT_ADDR),
+        &[
+            Coin {
+                denom: "ATOM".to_string(),
+                amount: Uint128::new(100_000),
+            },
+            Coin {
+                denom: "TIA".to_string(),
+                amount: Uint128::new(100_000),
+            },
+        ],
+    )]);
+
+    let env = mock_env();
+
+    let sender = "addr0000";
+
+    let info = mock_info(sender, &[]);
+    let msg = InstantiateMsg {
+        addresses: vec![
+            ("address0000".to_string(), Decimal::percent(60u64)),
+            ("address0001".to_string(), Decimal::percent(40u64)),
+        ],
+        cw20_contracts: vec![],
+    };
+
+    let fee_splitter_instance = instantiate(deps.as_mut(), env.clone(), info.clone(), msg).unwrap();
+    assert_eq!(
+        fee_splitter_instance.attributes,
+        vec![Attribute {
+            key: "initialized".to_string(),
+            value: "fee_splitter contract".to_string(),
+        }]
+    );
+
+    let msg = ExecuteMsg::SendTokens {
+        native_denoms: vec!["ATOM".to_string(), "TIA".to_string()],
+        cw20_addresses: vec!["cw20_contract_one".to_string()],
+    };
+
+    let res = execute(deps.as_mut(), env, info, msg).unwrap();
+
+    assert_eq!(
+        res.messages,
+        vec![
+            SubMsg {
+                id: 0,
+                msg: CosmosMsg::Bank(BankMsg::Send {
+                    to_address: "address0000".to_string(),
+                    amount: vec![
+                        Coin {
+                            denom: "ATOM".to_string(),
+                            amount: Uint128::new(60_000),
+                        },
+                        Coin {
+                            denom: "TIA".to_string(),
+                            amount: Uint128::new(60_000),
+                        }
+                    ]
+                }),
+                gas_limit: None,
+                reply_on: ReplyOn::Never
+            },
+            SubMsg {
+                id: 0,
+                msg: CosmosMsg::Bank(BankMsg::Send {
+                    to_address: "address0001".to_string(),
+                    amount: vec![
+                        Coin {
+                            denom: "ATOM".to_string(),
+                            amount: Uint128::new(40_000),
+                        },
+                        Coin {
+                            denom: "TIA".to_string(),
+                            amount: Uint128::new(40_000),
+                        }
+                    ]
+                }),
+                gas_limit: None,
+                reply_on: ReplyOn::Never
+            },
+        ]
+    );
+}
```

Comment:
You confirm sent messages but don't assert the target balances for the users.

Reply:
No, I meant the target balances. What we expect for the user's balance to be after the transaction is over.
But actually I realized, since we are only mocking the Banking module here then it's not needed. The proof of the `BankMsg::Sent` is good enough.

Reply:
But add similar test that will show correct CW20 transfer message.

Reply:
we have this message for both recipients:

```Rust
            SubMsg {
                id: 0,
                msg: CosmosMsg::Wasm(WasmMsg::Execute {
                    contract_addr: "asset0000".to_string(),
                    msg: to_json_binary(&Cw20ExecuteMsg::Transfer {
                        recipient: "address0000".to_string(),
                        amount: Uint128::new(60_000),
                    })
                    .unwrap(),
                    funds: vec![]
                }),
                gas_limit: None,
                reply_on: ReplyOn::Never
            },
```

would it be enough?